avalon/tools/cbloader/lib.py

```diff
@@ -30,25 +32,70 @@ def refresh_family_config():
     # Update the icons from the project configuration
     project = io.find_one({"type": "project"},
                           projection={"config.families": True})
-    if not project:
-        raise RuntimeError("Project not found")
+
+    assert project, "Project not found!"
     families = project['config'].get("families", [])
     families = {family['name']: family for family in families}
 
+    # Get all family states based on the config
+    family_states = get_family_filters(families)
     # Replace icons with a Qt icon we can use in the user interfaces
     default_icon = qtawesome.icon("fa.folder", color=FAMILY_ICON_COLOR)
     for name, family in families.items():
+        # Set family icon
         icon = family.get("icon", None)
         if icon:
             family['icon'] = qtawesome.icon("fa.{}".format(icon),
                                             color=FAMILY_ICON_COLOR)
         else:
             family['icon'] = default_icon
 
+        # Update state
+        family.update(family_states[name])
+
     # Default configuration
     families["__default__"] = {"icon": default_icon}
 
     FAMILY_CONFIG.clear()
     FAMILY_CONFIG.update(families)
 
     return families
+
+
+def get_family_filters(families):
+    """Set the family states based on host settings
+
+    This function has no effect if the lib module or the function in not present
+    in the registered config.
+
+    Args:
+        families (iterable): collection of family names, example:
+                             ["polly.imagesequence", "polly.camera_abc"]
+
+    Returns:
+        dict: collection of family settings
+    """
+
+    # Find host's lib module in the registered config
+    host = pipeline.registered_host()
+    host_name = host.__name__.rsplit(".", 1)[-1]
+
+    config = pipeline.registered_config()
+    config_name = config.__name__.split(".", 1)[0]
+    config_lib_module = "{}.{}.lib".format(config_name, host_name)
```

Comment:
More and more I'm feeling we're hacking our way into the config. This is finding it's way in too many places I'm suspecting.

We should try to keep at least as possible entry points into the config from avalon, and now we're opening so many doors.

Other ideas on how to do this?

Reply:
We have been provided with `registered_config` but not in the API. I have not seen it being used, probably because it has not been developed that far.

If we want to make this more accessible we can argue for a `pipeline.registered_config_name()`. We can also use the `Session["AVLAON_CONFIG"]` variable to retrieve the name and construct it from there. Might make it a bit less hacky. 

I do think the tools should also be moving forward in regards to development. Certainly if we are talking about implementing logic defined by the config for the tools. If we can create `before<TOOLNAME>Init` callbacks for each tool we might be able to move away our current "approach".